hack/sync_manifests.py

```diff
@@ -0,0 +1,184 @@
+import os
+import yaml
+import requests
+
+# Custom YAML dumper to preserve multi-line strings in block style, respect double quotes, and handle boolean-like strings.
+def string_representer(dumper, data):
+    # Check if the string is a boolean-like string ("true" or "false")
+    if str(data).lower() in ("true", "false"):
+        return dumper.represent_scalar("tag:yaml.org,2002:str", data, style='"')
+    # Check if the string is empty
+    if data == "":
+        return dumper.represent_scalar("tag:yaml.org,2002:str", data, style='"')
+    # Check if the string contains newlines
+    if "\n" in data:
+        return dumper.represent_scalar("tag:yaml.org,2002:str", data, style="|")
+    # Check if the string was originally quoted with double quotes
+    if hasattr(data, "style") and data.style == '"':
+        return dumper.represent_scalar("tag:yaml.org,2002:str", data, style='"')
+    return dumper.represent_scalar("tag:yaml.org,2002:str", data)
+
+yaml.add_representer(str, string_representer)
+
+# Get the latest version of Kueue.
+api_url = "https://api.github.com/repos/kubernetes-sigs/kueue/releases/latest"
```

Comment:
Can we make this script take a version?

ie python3 hack/sync_manifests.py 0.10.1?

I just want to make sure that we have deterministic behavior on the script and not to just always take the latest version.

We may have to provide patches for 0.10 kueue or 0.12 so we probably want to be able to generate bindata based on the version of kueue we provide.